### PR TITLE
Fragments in policies for cluster execution

### PR DESCRIPTION
**Pull request by gserranojc**
#### Description

Fragments moved to Serving-core. Fill fragments in SparktaClusterJob.
Paths in hdfs with id.
Bug for using TwitterJson in cluster execution corrected.
Mesos execution tested in the old cluster with front.

#### Test
Some test are moved from serving-api to serving-core

### CODE REVIEW

File: serving-core/src/main/scala/com/stratio/sparkta/serving/core/exception/ServingException.scala [hunk 1]
```diff
@@ -14,13 +14,13 @@
  * limitations under the License.
  */
```
Comment (danielcsant) [inline serving-core/src/main/scala/com/stratio/sparkta/serving/core/exception/ServingException.scala]: Could you change the name to ServingCoreException?

File: driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala [hunk 2]
```diff
@@ -44,30 +47,35 @@ object SparktaClusterJob extends SparktaSerializer {
     if (checkArgs(args)) {
       Try {
         initSparktaConfig(args(4), args(3))
```
Comment (anistal) [inline driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala]: Could you change this:
initSparktaConfig(args(4), args(3))

for this:
val ZookeperConfigurationIndex = 4

initiSparktaConfig(args(ZookeperConfigurationIndex), ... )

File: serving-core/pom.xml [hunk 3]
```diff
@@ -101,5 +101,15 @@
             <artifactId>mockito-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (anistal) [inline serving-core/pom.xml]: Is it necessary this dependency here?

File: serving-core/pom.xml [hunk 4]
```diff
@@ -101,5 +101,15 @@
...
+            <groupId>io.spray</groupId>
```
Comment (anistal) [inline serving-core/pom.xml]: Is it necessary this dependency here?

File: driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala (hunk 2 above)
Comment (sgomezg) [inline driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala]: If you want to separate arguments, I think this is a more clear approach:

```
val detailConfig = args(4)
val zkConfig = args(3)
initiSparktaConfig(detailConfig, zkConfig)
```

File: serving-core/pom.xml (hunk 4 above)
Comment (gserranojc) [inline serving-core/pom.xml]: Is necessary.

File: driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala (hunk 2 above)
Comment (anistal) [inline driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala]: That code do not pass our scalastyle.

Comment (sgomezg) [inline driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala]: Add index declaration as you said. My point is: maybe we want to declare specific variables to remind which is every argument.

Comment (sgomezg) [inline driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala]: One more thing. The correct way may be to use named arguments, but our target must be to remove spark-submit as command call. IMHO :)

File: serving-api/src/main/scala/com/stratio/sparkta/serving/api/actor/ClusterSparkStreamingContextActor.scala [hunk 5]
```diff
@@ -67,19 +68,16 @@ class ClusterSparkStreamingContextActor(policy: AggregationPoliciesModel,
...
-          val hdfsUtils = new HdfsUtils(hdfsUgi,hdfsConf)
+          val hdfsConf = HdfsUtils.hdfsConfiguration(hadoopConfDir)
+          val hdfsUtils = new HdfsUtils(hdfsUgi, hdfsConf)
           val pluginsJarsFiles = PolicyUtils.activeJarFiles(activeJars.right.get, jarsPlugins)
           val pluginsJarsPath =
-            s"/user/$hadoopUserName/${policy.name}-${policy.id.get}/${
-              hdfsConfig.getString(AppConstant
-                .PluginsFolder)
-            }/"
+            s"/user/$hadoopUserName/${policy.id.get.trim}/${hdfsConfig.getString(AppConstant.PluginsFolder)}/"
```
Comment (arinconstrio) [inline serving-api/src/main/scala/com/stratio/sparkta/serving/api/actor/ClusterSparkStreamingContextActor.scala]: can we trim the policy id only once?

File: driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala (hunk 2 above)
Comment (gserranojc) [inline driver/src/main/scala/com/stratio/sparkta/driver/SparktaClusterJob.scala]: I had the same code that Sergio says, but as the method is to big and clear. If I change it as Alvaro says and I can remove the magic numbers.

I make one small refactor and submmit changes

File: serving-api/src/main/scala/com/stratio/sparkta/serving/api/actor/ClusterSparkStreamingContextActor.scala (hunk 5 above)
Comment (gserranojc) [inline serving-api/src/main/scala/com/stratio/sparkta/serving/api/actor/ClusterSparkStreamingContextActor.scala]: Ok! I change this.

**Comment by danielcsant** [comment]
LGTM

**Pull request merged**